crates/payload/basic/src/lib.rs

```diff
@@ -547,6 +597,22 @@ struct PayloadConfig {
     attributes: PayloadBuilderAttributes,
     /// The chain spec.
     chain_spec: Arc<ChainSpec>,
+    /// The rollup's compute pending block configuration option.
+    /// TODO(clabby): Implement this feature.
+    #[cfg(feature = "optimism")]
+    #[allow(dead_code)]
+    compute_pending_block: bool,
+}
+
+impl PayloadConfig {
+    /// Returns an owned instance of the [PayloadConfig]'s extra_data bytes.
+    pub(crate) fn extra_data(&self) -> reth_primitives::Bytes {
+        #[cfg(feature = "optimism")]
+        if self.chain_spec.optimism {
+            return Default::default()
+        }
+        reth_primitives::Bytes(self.extra_data.clone())
+    }
```

Comment:
can we replace extra data earlier ... in `BasicPayloadJobGeneratorConfig` maybe? `PayloadConfig` is usually destructured and it's not obvious that extra data should be retrieved this way